src/client/src/App.js

```diff
@@ -1,33 +1,140 @@
-import React from "react";
+import React from 'react'
+
 import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
 
-import Header from "./components/Header";
+import Header, {AdminHeader} from "./components/Header";
 
 import HomePage from './pages/Home';
 import Admin from './pages/Admin';
 import DataView from './pages/DataView360/DataView360';
 import About from './pages/About';
+import Login from './components/Login/Login';
+import Check from './pages/Check/Check';
+import useToken from './components/Login/useToken';
+var jwt = require('jsonwebtoken');
+
+// Triggers token expiration check 
+const sleep = time => new Promise(resolve => setTimeout(resolve, time))
+const expTimer = () => sleep(500).then(() => ({})) 
+
+const AuthContext = React.createContext()
+
+function AuthProvider({children}) {
+  const [state, setState] = React.useState({
+    status: 'pending',
+    error: null,
+    user: null,  })
+
+  React.useEffect(() => {
+    expTimer().then(
+      user => setState({status: 'success', error: null, user})  // 
+    )
+  }, )
+
+  return (
+    <AuthContext.Provider value={state}>
+        {state.status === 'pending' ? (
+        'App ACP: Loading...'
+      ) : state.status === 'error' ? (
+        <div>
+          Oh no
+          <div>
+            <pre>{state.error.message}</pre>
+          </div>
+        </div>
+      ) : (
+        children
+      )}
+    </AuthContext.Provider>
+  )
+}
+
+function useAuthState() {
+  const state = React.useContext(AuthContext)
+  const isPending = state.status === 'pending'
+  const isError = state.status === 'error'
+  const isSuccess = state.status === 'success'
+  const isAuthenticated = state.user && isSuccess
+  return {
+    ...state,
+    isPending,
+    isError,
+    isSuccess,
+    isAuthenticated,
+  }
+}
+
+
+function AuthenticatedApp() {
+
+    const { access_token, setToken } = useToken();
+
+
+    var decoded = jwt.decode(access_token, { complete: true });
+    const userName = decoded?.payload.sub;
+    const userRole = decoded?.payload.role;
+    var expTime =  decoded?.payload.exp -  Date.now()/1000;
```

Comment:
better to use moment here

Reply:
What advantage is there over the JS built-in [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)?

Reply:
it deals better with exceptions such as nulls. also it has built in functions like toLocal etc that will be easier to manage if we want to make changes later. 
Also since we are already using it in multiple places, better to be consistant for new developers sake

Reply:
I've left this for now due to weird behavior with moment.